src/benchq/algorithms/sel_block_encoding.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+from orquestra.quantum.circuits import Circuit, RZ, PHASE, SX, X
+
+
+
+def coef_k(k:int, beta) -> "Circuit":
+    """Constructs unitaries that prepare states COEF_k and COEF_k_prime corresponding to the z_k. 
+
+        Args:
+            k (int): The numper of the grid points to approximate the countour integral.
+            beta (float?): bounding parameter of the matrix.
+
+        Returns: 
+            prep, prep_prime (Circuit): unitary corresponding to PREP and PREP_prime
+    """
+
+    alpha = beta / 2.0
+    theta = np.arcsin(np.sqrt((alpha) / (beta+alpha))) #theta/2
+    prep = Circuit()
+    prep_prime = Circuit()
+    
+    for grid_point in range(k):
+        phi = np.pi * grid_point / k
+        
+        # decompose using U3
+        prep += PHASE(-3*phi/2 - theta/2 - np.pi/2)(grid_point)
+        prep += RZ(-phi + np.pi)(grid_point) #P gate
+        prep += SX.dagger(grid_point)
+        prep += RZ(2*theta + np.pi)(grid_point)
+        prep += SX(grid_point)
+        
+        prep_prime += PHASE(-3*phi/2 - theta/2 - np.pi/2)(grid_point)
+        prep_prime += RZ(phi + np.pi)(grid_point)
+        prep_prime += SX.dagger(grid_point) 
+        prep_prime += RZ(-2*theta + np.pi)(grid_point)
+        prep_prime += SX(grid_point)
+    
+    return prep, prep_prime
+
+
+def control_prep(k:int, beta):
+    """Constructs the control-PREP and control-PREP_prime unitaries. 
+
+        Args:
+            k (int): The numper of the grid points to approximate the countour integral.
+            beta (float?): bounding parameter of the matrix.
+
+        Returns: 
+            [placeholder]
+    """
+
+    # should there be a check for k, k>0??
+    alpha = beta / 2.0
+    theta = np.arcsin(np.sqrt((alpha) / (beta+alpha))) #theta/2
+    n_qubits = int(np.ceil(np.log2(k)))
+    control_qubits = range(n_qubits)
+    prep = Circuit()
+    prep_prime = Circuit()
+    
+    for grid_point in range(k):
+        phi = np.pi * grid_point / k
+        
+        binary = bin(grid_point)[2::]
+        reversed_bin = binary[::-1]
+        ones = [indx for indx, digit in enumerate(reversed_bin) if digit == '1']
+        # zeroes = [indx for indx, digit in enumerate(reversed_bin) if digit == '0']
```

Comment:
try to leave out commented code which doesn't tell other programs what this is doing.